src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -137,10 +137,16 @@ try
 {
     const auto drawingContext = static_cast<const DrawingContext*>(clientDrawingContext);
 
-    const DWRITE_FONT_WEIGHT weight = _fontRenderData->DefaultFontWeight();
+    DWRITE_FONT_WEIGHT weight = _fontRenderData->DefaultFontWeight();
     DWRITE_FONT_STYLE style = _fontRenderData->DefaultFontStyle();
     const DWRITE_FONT_STRETCH stretch = _fontRenderData->DefaultFontStretch();

```

Comment:
Is there anything that can change the `stretch` property? Is there a SGR sequence for this? Or this is just gonna sit here not being used.

Reply:
Poking around the various standards...

[T.416](https://www.itu.int/rec/T-REC-T.416-199303-I/en) and [ECMA-48](https://www.ecma-international.org/publications-and-standards/standards/ecma-48/) document some CSI sequences for character spacing and character separation, but that's not really the same thing, as I suspect they expected extra space in the layout, not stretching the characters.

Select Character Spacing - SHS (`CSI Pn <space>K`) is probably the closest conceptually, since it lets you pick from a width enum 0..6, per ECMA-48. T.416 only includes 0..4, and measures them in BMU instead of millimeters, but with the same relative meanings. [T.61](https://www.itu.int/rec/T-REC-T.61) uses the same definition as ECMA-48, but notes that `3` is only for Chinese ideogram and Japanese Kanji terminals, and `5` and `6` are only for Chinese ideogram terminals, suggesting that they relate to things like the "full-width" latin character layout from Japanese fonts, [i.e. a latin character stretched to take the same width as a kanji character](http://xahlee.info/comp/unicode_full-width_chars.html) and whatever the similar concept is in Chinese typography.

* 0: 10 characters per 25,4 mm 
* 1: 12 characters per 25,4 mm 
* 2: 15 characters per 25,4 mm 
* 3: 6 characters per 25,4 mm 
* 4: 3 characters per 25,4 mm 
* 5: 9 characters per 50,8 mm 
* 6: 4 characters per 25,4 mm

Anyway, I doubt it's ever used, and if it is, presumably that use-case will also be interested in Set Character Spacing - SCS which lets you specify the value rather than pick from an enum, and unlike width, we can't pass random values outside the [`DWRITE_FONT_STRETCH` enum](https://docs.microsoft.com/en-us/windows/win32/api/dwrite/ne-dwrite-dwrite_font_stretch) to DirectWrite here, so it'd have to be done with something like [`DWRITE_FONT_AXIS_VALUE`](https://docs.microsoft.com/en-us/windows/win32/api/dwrite_3/ns-dwrite_3-dwrite_font_axis_value)'s "Width" support.

Assuming 100 means the same as enum value 0 above, that might actually be feasible, but it's a newer version of DWrite (`dwrite_3.h`) than we're apparently using here, using [`IDWriteFontFamily2::GetMatchingFonts`](https://docs.microsoft.com/en-us/windows/win32/api/dwrite_3/nf-dwrite_3-idwritefontfamily2-getmatchingfonts) instead of [`IDWriteFontFamily::GetFirstMatchingFont`](https://github.com/microsoft/terminal/blob/85c485e94fd067ab6541c1b8c7f6d61b33099d58/src/renderer/dx/DxFontInfo.cpp#L241-L242), which needs [Build 16299: Windows 10 Fall Creators Update](https://docs.microsoft.com/en-au/windows/win32/directwrite/what-s-new-in-directwrite-for-windows-8-consumer-preview#windows-10-fall-creators-update).

But it does suggest that somewhere there's a mapping from the `DWRITE_FONT_STRETCH` values to the `DWRITE_FONT_AXIS_VALUE` range, if we do want to try and implement SHS at some far-far-far-far-future point.